packages/scoped-custom-element-registry/src/scoped-custom-element-registry.ts

```diff
@@ -11,6 +11,61 @@
  * subject to an additional IP rights grant found at
  * http://polymer.github.io/PATENTS.txt
  */
+
+interface CustomElementConstructor {
+  observedAttributes?: Array<string>;
+  formAssociated?: boolean;
+
+  new (...params: unknown[]): CustomHTMLElement;
+}
+
+interface CustomHTMLElement {
+  connectedCallback?(): void;
+  disconnectedCallback?(): void;
+  adoptedCallback?(): void;
+  attributeChangedCallback?(
+    name: string,
+    oldValue?: string | null,
+    newValue?: string | null,
+    namespace?: string | null
+  ): void;
+  formAssociatedCallback?(form: HTMLFormElement | null): void;
+  formDisabledCallback?(disabled: boolean): void;
+  formResetCallback?(): void;
+  formStateRestoreCallback?(
+    state: File | string | FormData | null,
+    mode: string
+  ): void;
+}
+
+interface CustomElementRegistry {
+  _getDefinition(tagName: string): CustomElementDefinition;
+}
+
+interface CustomElementDefinition {
+  elementClass: CustomElementConstructor;
+  connectedCallback?: CustomHTMLElement['connectedCallback'];
```

Comment:
```suggestion
  /**
   * We hold onto the versions of callbacks at registration time, because that's the
   * specc'd behavior.
   */
  connectedCallback?: CustomHTMLElement['connectedCallback'];
```

Reply:
Done.